src/assets/js/main.js

```diff
@@ -85,20 +85,55 @@ function scrollSidebarIntoView() {
   }
 }
 
-function adjustToc() {
-  // Adjustments to the jekyll-toc TOC.
+/**
+ * Adjusts the behavior of the table of contents (TOC) on the page.
+ * 
+ * This function enables a "scrollspy" feature on the TOC, where the active link in the TOC is updated
+ * based on the currently visible section in the page.
```

Comment:
To shorten this a bit:

```suggestion
 * This function enables a "scrollspy" feature on the TOC, 
 * where the active link in the TOC is updated
 * based on the currently visible section in the page.
```